src/circuits/atomicV3.ts

```diff
@@ -0,0 +1,124 @@
+import { PubSignalsVerifier, VerifyOpts } from '@lib/circuits/registry';
+import { checkQueryRequest, ClaimOutputs, Query } from '@lib/circuits/query';
+import { Resolvers } from '@lib/state/resolver';
+import { IDOwnershipPubSignals } from '@lib/circuits/ownershipVerifier';
+import { checkIssuerNonRevState, checkUserState, getResolverByID } from '@lib/circuits/common';
+import { DID, getDateFromUnixTimestamp } from '@iden3/js-iden3-core';
+import { DocumentLoader } from '@iden3/js-jsonld-merklization';
+import { AtomicQueryV3PubSignals, byteEncoder, JSONObject, ProofType } from '@0xpolygonid/js-sdk';
+
+const valuesSize = 64;
+const defaultProofVerifyOpts = 1 * 60 * 60 * 1000; // 1 hour
+
+export class AtomicQueryV3PubSignalsVerifier
+  extends IDOwnershipPubSignals
+  implements PubSignalsVerifier
+{
+  pubSignals = new AtomicQueryV3PubSignals();
+
+  constructor(pubSignals: string[]) {
+    super();
+    this.pubSignals = this.pubSignals.pubSignalsUnmarshal(
+      byteEncoder.encode(JSON.stringify(pubSignals))
+    );
+
+    this.userId = this.pubSignals.userID;
+    this.challenge = this.pubSignals.requestID;
+  }
+
+  async verifyQuery(
+    query: Query,
+    schemaLoader?: DocumentLoader,
+    verifiablePresentation?: JSON,
+    opts?: VerifyOpts,
+    params?: JSONObject
+  ): Promise<void> {
+    const outs: ClaimOutputs = {
+      issuerId: this.pubSignals.issuerID,
+      schemaHash: this.pubSignals.claimSchema,
+      slotIndex: this.pubSignals.slotIndex,
+      operator: this.pubSignals.operator,
+      value: this.pubSignals.value,
+      timestamp: this.pubSignals.timestamp,
+      merklized: this.pubSignals.merklized,
+      claimPathKey: this.pubSignals.claimPathKey,
+      claimPathNotExists: this.pubSignals.claimPathNotExists,
+      valueArraySize: valuesSize,
+      isRevocationChecked: this.pubSignals.isRevocationChecked
+    };
+    await checkQueryRequest(query, outs, schemaLoader, verifiablePresentation, opts);
+
+    const { proofType, verifierID, nullifier, nullifierSessionID } = this.pubSignals;
+
+    const isValidSigType = query.proofType === ProofType.BJJSignature && proofType === 1;
+    const isValidMTPType =
+      query.proofType === ProofType.Iden3SparseMerkleTreeProof && proofType === 2;
+
+    if (!isValidSigType && !isValidMTPType) {
```

Comment:
this is wrong check.
if proofType == '' , then signal can be 1 or 2. 
see https://github.com/iden3/go-iden3-auth/blob/8e8050511faf8384126ae6bb44641a5611b8e888/pubsignals/atomicV3.go#L57